File: FindAssemblies_recursive_prepruned.m
function [assemD]=FindAssemblies_recursive_prepruned(binMaus,w1,w2,maxlag,Dc,ref_lag)
%  © 2020 Russo
%  for information please contact eleonora.russo@zi-mannheim.de

   
%% zero order  
assembly_in.n{1}.elements=w1;
assembly_in.n{1}.lag=[];
assembly_in.n{1}.pr=[];
assembly_in.n{1}.Time=binMaus(1,:)';
assembly_in.n{1}.Noccurrences=sum(binMaus(1,:));

assembly_in.n{2}.elements=w2;
assembly_in.n{2}.lag=[];
assembly_in.n{2}.pr=[];
assembly_in.n{2}.Time=binMaus(2,:)';
assembly_in.n{2}.Noccurrences=sum(binMaus(2,:));
    
%% first order: test over pairs

[assemD]=TestPair_ref(assembly_in.n{1},binMaus(2,:)',w2,maxlag,Dc,ref_lag);



end


      

File: python_CADopti/FindAssemblies_recursive_prepruned.py
import numpy as np
from typing import Dict, List, Any

def find_assemblies_recursive_prepruned(bin_maus: np.ndarray, w1: int, w2: int, max_lag: int, dc: float, ref_lag: int) -> Dict[str, Any]:
    """
    Python translation of FindAssemblies_recursive_prepruned function.
    
    :param bin_maus: Binary matrix of assembly units
    :param w1: First assembly element
    :param w2: Second assembly element
    :param max_lag: Maximum lag to consider
    :param dc: Threshold for significance
    :param ref_lag: Reference lag
    :return: Dictionary containing assembly information
    """
    
    # Zero order
    assembly_in = {
        'n': [
            {
                'elements': w1,
                'lag': [],
                'pr': [],
                'Time': bin_maus[0, :].T,
                'Noccurrences': np.sum(bin_maus[0, :])
            },
            {
                'elements': w2,
                'lag': [],
                'pr': [],
                'Time': bin_maus[1, :].T,
                'Noccurrences': np.sum(bin_maus[1, :])
            }
        ]
    }
    
    # First order: test over pairs
    assem_d = test_pair_ref(assembly_in['n'][0], bin_maus[1, :].T, w2, max_lag, dc, ref_lag)
    
    return assem_d

# Note: You'll need to implement the TestPair_ref function (renamed to test_pair_ref in Python)
# as it's not provided in the original MATLAB code snippet.

def test_pair_ref(assembly1: Dict[str, Any], bin_maus2: np.ndarray, w2: int, max_lag: int, dc: float, ref_lag: int) -> Dict[str, Any]:
    """
    Placeholder for the TestPair_ref function. Implement the logic here.
    """
    # Implement the TestPair_ref logic here
    pass